deploy/kubernetes/images/stable/image.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-csi-rpovisioner
+imageTag:
+  name: gke.gcr.io/csi-provisioner
+  newTag: "v1.6.0-gke.0"
+
+---
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-csi-attacher
+imageTag:
+  name: gke.gcr.io/csi-attacher
+  newTag: "v2.2.0-gke.0"
+---
+
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-csi-resizer
+imageTag:
+  name: gke.gcr.io/csi-resizer
+  newTag: "v0.5.0-gke.0"
+---
+
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-csi-snapshotter
+imageTag:
+  name: gke.gcr.io/csi-snapshotter
+  newTag: "v2.1.1-gke.0"
+---
+
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-csi-node-registrar
+imageTag:
+  name: gke.gcr.io/csi-node-driver-registrar
+  newTag: "v1.3.0-gke.0"
+---
+
+apiVersion: builtin
+kind: ImageTagTransformer
+metadata:
+  name: imagetag-gcepd-driver
+imageTag:
+  name: gke.gcr.io/gcp-compute-persistent-disk-csi-driver
+  # Don't change stable image without changing pdImagePlaceholder in
```

Comment:
Let me think out loud briefly. I haven't used kustomize much so I could be missing something.

Do we want the yaml in base/ to be meaningful, or is it only supported by deploying through customize? In that case we could replace the images in base/ with placeholders like ATTACHER_IMAGE and have a newName in all of the image transformers. Then it's clear what is being changed.

With respect to the ordering of multiple changes, though, it looks like there's a bug in kustomize. If one applies two patches that refer to the same field, kustomize errors. But if one applies multiple image transformations, the last one wins (as defined by the list in the kustomization.yaml in the transformer directory).

So that's unfortunate... I can ask in the kustomize repo to see if that really is a bug or for some reason WAI. From our perspective, we just have to be careful I guess, and trace the order of things from kustomization files starting in the overlay.

Reply:
I think using placeholders like ATTACHER_IMAGE sounds good to me. All the image name placeholder used in /base should not be touched.  
All the image replacement will be done in /images dir. For testing, we only need to touch/replace /images/dev/image.yaml file to create ImageTagTransformer when necessary.

Reply:
Also I suggest to modify placeholder in a different PR. This PR has already gone through lots of changes.

Reply:
SGTM for doing that in a different PR

Reply:
I've filed https://github.com/kubernetes-sigs/kustomize/issues/2590 about conflict with image transformers.

Reply:
Thanks! I cc @Liujingfang1 who is the expert in this area.